packages/connect/src/connector/index.ts

```diff
@@ -0,0 +1,15 @@
+import { getConnectorClient as slGetConnectorClient } from "./smoldot-light.js"
+import { getConnectorClient as eGetConnectorClient } from "./extension.js"
+import { DOM_ELEMENT_ID } from "@substrate/connect-extension-protocol"
+
+export * from "./errors.js"
+export * from "./types.js"
+
+const isExtension =
+  typeof document === "object" &&
+  typeof document.getElementById === "function" &&
+  !!document.getElementById(DOM_ELEMENT_ID)
+
+export const getConnectorClient = isExtension
+  ? eGetConnectorClient
+  : slGetConnectorClient
```

Comment:
One thing that this approach doesn't handle is that if the extension returns errors, which could happen for example because it has reached the maximum number of chains, then we should probably fall back to the `SmoldotProvider`.

Reply:
Yeah, I agree that would be a nice feature that have. However, we currently don't have this behavior and it's a bit tricky both to implement and to get the right API for it. So, I would rather tackle that in the future, once we know how we want to handle that, and how much the consumer should be aware of what's happening...